ignite/metrics/classification_report.py

```diff
@@ -0,0 +1,142 @@
+import json
+from typing import Callable, Collection, Dict, List, Optional, Sequence, Union
+
+import torch
+
+from ignite.metrics.fbeta import Fbeta
+from ignite.metrics.metric import Metric, reinit__is_reduced
+from ignite.metrics.metrics_lambda import MetricsLambda
+from ignite.metrics.precision import Precision
+from ignite.metrics.recall import Recall
+
+__all__ = ["ClassificationReport"]
+
+
+class ClassificationReport(MetricsLambda):
```

Comment:
Previously, we used methods to create composed metrics like this one. For example, FBeta, IoU, mIoU etc.
Do you think we can define `ClassificationReport` as a method instead of deriving from `MetricsLambda`. It seems to me that inheritance here is very artificial...

Reply:
I agree on that point, my only issue with having a method returning the metrics lambda is with the update method. Metrics lambda update won't update underlying metrics automatically. it worked in the fbeta because we were computing final averages. I am not sure if it'd be possible here.

Reply:
Yes, that's true however there are two points here:
- MetricLambda does not work as user would expect with reset/update/compute API : 
https://github.com/pytorch/ignite/blob/f9504f1cea9783508d7ff6d852588167b2f4fa6f/ignite/metrics/metrics_lambda.py#L82-L87

It expects to be attached to the engine or to call update on ingredients. You are right, we can not call update on precision/recall/fbeta separately. This would be something to work on in another PR. If you could create an issue for that, I'd really appreciate.

- let's make this metric as a method and make it work with attach to engine API. Once `MetricLambda` can work with reset/update/compute API, this will be automatically inherited ...